pkg/tsdb/influxdb/influxdb.go

```diff
@@ -96,17 +96,13 @@ func (e *InfluxDBExecutor) Query(ctx context.Context, dsInfo *models.DataSource,
 }
 
 func (e *InfluxDBExecutor) getQuery(dsInfo *models.DataSource, queries []*tsdb.Query, context *tsdb.TsdbQuery) (*Query, error) {
-	for _, v := range queries {
-
-		query, err := e.QueryParser.Parse(v.Model, dsInfo)
-		if err != nil {
-			return nil, err
-		}
-
-		return query, nil
+	// The model supports multiple queries, but right now this is only used from
+	// alerting so we only needed to support batch executing 1 query at a time.
+	query, err := e.QueryParser.Parse(queries[0].Model, dsInfo)
```

Comment:
This still feels like a change for the worse. The old code would not throw index out of range panic for empty queries arrays where with this change it would. If you remove the range loop you need to add a length check and return the error that the old code did when no queries found

Reply:
@torkelo  No problem. I dropped this change, so that I will be fixing it later on, adding Megacheck to the CI.